### PR TITLE
[MIG][9.0][sale_packaging_price] Migrate to v9.

### PR DESCRIPTION
**Pull request by yajo**
Migrate to v9.
- Some view & menu IDs have been removed or changed.
- LGPL, with original author permission.

@Tecnativa #223

### CODE REVIEW

File: sale_packaging_price/README.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+rounding issues to get the exact price that are warned if happens. It also adds
+a field to set the package material.
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+#. Go to *Warehouse > Configuration > Settings*.
+#. Enable *Packages > Traceability > Record packages used on packing: pallets,
```
Comment (rafaelbn) [inline sale_packaging_price/README.rst]: Enable _Traceability > Packager > Record packager used on packing:pallets, boxes, ..._

File: sale_packaging_price/README.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+unit according the number of pieces that fit in that packaging, so there can be
+rounding issues to get the exact price that are warned if happens. It also adds
+a field to set the package material.
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+#. Go to *Warehouse > Configuration > Settings*.
```
Comment (rafaelbn) [inline sale_packaging_price/README.rst]: s/Warehouse/Inventory

**Comment by rafaelbn** [comment]
Thanks, some comments but :+1:

File: sale_packaging_price/models/product.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl
+
+from openerp import api, fields, models, _
+from openerp.addons.decimal_precision import decimal_precision as dp
+
+
+class ProductPackagingMaterial(models.Model):
```
Comment (antespi) [inline sale_packaging_price/models/product.py]: This should be split in other file named `product_packaging_material.py`

File: sale_packaging_price/models/product.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    _name = 'product.packaging.material'
+
+    name = fields.Char(string='Package Material')
+
+
+class ProductPackaging(models.Model):
+    _inherit = 'product.packaging'
```
Comment (antespi) [inline sale_packaging_price/models/product.py]: Filename should be `product_packaging.py`

File: sale_packaging_price/models/sale_order.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl
+
+from openerp import models, fields, api
+from openerp.addons.decimal_precision import decimal_precision as dp
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
```
Comment (antespi) [inline sale_packaging_price/models/sale_order.py]: Filename should be `sale_order_line.py`

File: sale_packaging_price/tests/test_sale_packaging_price.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        self.sale_order.order_line.product_packaging = self.product_pack_6
+        self.sale_order.order_line._onchange_product_packaging()
+        self.assertEqual(self.sale_order.order_line.price_unit, 30)
+        self.assertEqual(self.sale_order.order_line.packaging_weight, 16.5)
+
+    def test_onchange_price_list(self):
+        """User is warned when pack price is not divisible by qty."""
+        self.product_pack_6.list_price = 100
+        res = self.product_pack_6._onchange_list_price()
+        self.assertIn('warning', res)
```
Comment (antespi) [inline sale_packaging_price/tests/test_sale_packaging_price.py:61]: You can use this form, maybe it is more explicit:

```
from openerp.exceptions import Warning
(...)
with self.assertRaises(Warning):
    res = self.product_pack_6._onchange_list_price()
```

Comment (pedrobaeza) [inline sale_packaging_price/tests/test_sale_packaging_price.py:61]: This is not a exception, but a warning returned in a dictionary

File: sale_packaging_price/README.rst [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+#. Enable *Traceability > Packages > Record packages used on packing: pallets,
+   boxes, ...*
+#. Enable *Products > Packaging Methods > Manage available packaging options
+   per products*.
+#. *Apply*.
+
+Usage
+=====
+
+First, ensure you have permissions for *Technical Settings / Manage Product
```
Comment (antespi) [inline sale_packaging_price/README.rst:30]: I can't find this permission in Administrator user form and access rights tab:

image

File: sale_packaging_price/README.rst [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+Then, to use this module, you need to:
+
+#. Go to *Inventory > Inventory Control > Products*.
+#. Edit or create a product.
+#. Go to *Inventory* tab.
+#. Under *Packaging* you can create packages for this product.
+
+To batch managing of packages, you need to:
+
+* Go to *Inventory > Inventory Control > Packagings*.
```
Comment (antespi) [inline sale_packaging_price/README.rst:42]: Do you mean `Inventory Control > Product Packaging` or `Inventory Control > Packages`?

File: sale_packaging_price/models/product.py (hunk 3 above)
Comment (yajo) [inline sale_packaging_price/models/product.py]: It seems the main model here is product.packaging, so I will apply guidelines and rename to `product_packaging.py`, but it seems not splitting is justified this time.

File: sale_packaging_price/tests/test_sale_packaging_price.py (hunk 6 above)
Comment (yajo) [inline sale_packaging_price/tests/test_sale_packaging_price.py:61]: Yes, although what @antespi says could be done, but with no clear guidelines, I'll leave it as it is now.

File: sale_packaging_price/__openerp__.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
```
Comment (carlosdauden) [inline sale_packaging_price/__openerp__.py:3]: Please change headers to  `© 2015 Carlos Dauden <carlos.dauden@tecnativa.com>`

**Comment by carlosdauden** [comment]
:+1:

**Comment by rafaelbn** [comment]
Thanks :+1:

**Comment by rafaelbn** [comment]
Sorry @carlosdauden I miss you comment. I make a PR for this...

**Pull request merged**